### PR TITLE
ENH: Use metadata to populate a default README

### LINKED ISSUE #1989: Enhance `add_readme` plugin to pull basic dataset info from metadata
Right now we just add some boilerplate message, could be made much more useful with minimal effort.

### PR DESCRIPTION
**Pull request by mih**
For studyforrest (https://github.com/psychoinformatics-de/studyforrest-data-phase2) this looks something like below. Now done with markdown for simplicity and compatibility with other kind of markup. RST blends relatively nicely with MD.

This fixes #1989 

--------

# Dataset "studyforrest_phase2"

Extension of the dataset published in Hanke et al. (2014; doi:10.1038/sdata.2014.3) with additional acquisitions for 15 of the original 20 particpants. These additions include: retinotopic mapping, a localizer paradigm for higher visual areas (FFA, EBA, PPA), and another 2h movie recording with 3T full-brain BOLD fMRI with simultaneous 1000 Hz eyetracking.

### Authors

- Michael Hanke
- Falko R. Kaule
- Ayan Sengupta
- Florian J. Baumgartner
- J. Swaroop Guntupalli
- Christian Häusler
- Michael Hoffmann
- Vittorio Iacovella
- Daniel Kottke
- Jörg Stadler

### Reference

http://studyforrest.org

### License

PDDL

### Funding

We acknowledge the support of the Combinatorial NeuroImaging Core Facility at the Leibniz Institute for Neurobiology in Magdeburg, and the German federal state of Saxony-Anhalt, Project: Center for Behavioral Brain Sciences. This research was, in part, also supported by the German Federal Ministry of Education and Research (BMBF) as part of a US-German collaboration in computational neuroscience (CRCNS), co-funded by the BMBF and the US National Science Foundation (BMBF 01GQ1112; NSF 1129855). Work on the data-sharing technology employed for this research was supported by US-German CRCNS project, co-funded by the BMBF and the US National Science Foundation (BMBF 01GQ1411; NSF 1429999).

## General information

This is a DataLad dataset (id: 6871e0a6-be6f-11e7-8d7f-a0369fb55db0).

For more information on DataLad and on how to work with its datasets,
see the DataLad documentation at: http://docs.datalad.org

### CODE REVIEW

File: datalad/plugin/add_readme.py [hunk 1]
```diff
@@ -56,11 +50,51 @@ def dlplugin(dataset, filename='README.rst', existing='skip'):
...
+            ('Homepage', meta.get('homepage', '')),
+            ('Reference', meta.get('citation', '')),
+            ('License', meta.get('license', '')),
+            ('Keywords', u', '.join([u'`{}`'.format(k) for k in assure_list(meta.get('tag', []))])),
+            ('Funding', meta.get('fundedby', '')),
+            ):
+        if label and content:
+            metainfo += u'\n\n### {}\n\n{}'.format(label, content)
+        elif content:
+            metainfo += u'\n\n{}'.format(content)
```
Comment (yarikoptic) [inline datalad/plugin/add_readme.py]: apparently no test triggers any of the above lines enhancing metainfo.  Would be nice to have at least a very basic test to see if formatting looks ok

File: datalad/plugin/add_readme.py [hunk 2]
```diff
@@ -56,11 +50,51 @@ def dlplugin(dataset, filename='README.rst', existing='skip'):
...
+
+For more information on DataLad and on how to work with its datasets,
+see the DataLad documentation at: http://docs.datalad.org
+""".format(
+        title='Dataset "{}"'.format(meta['name']) if 'name' in meta else 'About this dataset',
+        metainfo=metainfo,
+        id=u' (id: {})'.format(dataset.id) if dataset.id else '',
+        )
+
+    print(meta)
```
Comment (yarikoptic) [inline datalad/plugin/add_readme.py]: is print intended or a leftover?

Comment (mih) [reply datalad/plugin/add_readme.py]: Thanks for catching that!

Comment (mih) [reply datalad/plugin/add_readme.py]: Removed.

File: datalad/plugin/add_readme.py (hunk 1 above)
Comment (mih) [reply datalad/plugin/add_readme.py]: Implemented.

**Comment by mih** [comment]
Nice, github is stuck reporting a conflict that got resolved using its own web editor...

**Comment by yarikoptic** [comment]
oooo -- you are signing commits/merges now? I should start doing the same. Did only for release tags so far

**Pull request merged**